controller/controller.go

```diff
@@ -1570,34 +1570,27 @@ func (ctrl *ProvisionController) deleteVolumeOperation(ctx context.Context, volu
 	return nil
 }
 
+// removeFinalizer removes finalizer from slice, returns slice and whether modified.
 func removeFinalizer(finalizers []string, finalizerToRemove string) ([]string, bool) {
-	modified := false
-	modifiedFinalizers := make([]string, 0)
-	for _, finalizer := range finalizers {
-		if finalizer != finalizerToRemove {
-			modifiedFinalizers = append(modifiedFinalizers, finalizer)
+	for i, finalizer := range finalizers {
+		if finalizer == finalizerToRemove {
+			return append(finalizers[:i], finalizers[i+1:]...), true
```

Comment:
This should return `nil` instead of empty array to fix the unit tests.

Reply:
Updated, PTAL thanks.